File: src/main/java/ru/romangr/catbot/telegram/model/Message.java
package ru.romangr.catbot.telegram.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
import lombok.Builder;
import lombok.Value;

/**
 * Roman 27.10.2016.
 */
@Value
@Builder
@JsonDeserialize(builder = Message.MessageBuilder.class)
public class Message {
    @JsonProperty("message_id")
    public int id;

    public User from;

    public Chat chat;

    public String text;

    @JsonPOJOBuilder(withPrefix = "")
    public static class MessageBuilder {

        @JsonProperty("message_id")
        public MessageBuilder id(int id) {
            this.id = id;
            return this;
        }

    }
}

File: src/main/java/ru/romangr/catbot/telegram/model/Message.kt
package ru.romangr.catbot.telegram.model

import com.fasterxml.jackson.annotation.JsonProperty
import ru.romangr.catbot.telegram.dto.Document
import ru.romangr.catbot.telegram.dto.Video

/**
 * Roman 27.10.2016.
 */
data class Message(
        @JsonProperty("message_id") val id: Long,
        val from: User,
        val chat: Chat,
        val text: String? = null,
        val video: Video? = null,
        val document: Document? = null,
        val photo: List<PhotoSize>? = null,
        val caption: String? = null
)

data class PhotoSize(
        @JsonProperty("file_id") var fileId: String,
        var width: Long
)